core/parentchain/light-client/src/io.rs

```diff
@@ -19,7 +19,8 @@ use crate::{
 	error::Result,
```

Comment:
Here I mainly changed the initialization flow, which was really hard to handle because of side-effects of some functions. I think that there is still very much potential, but the code will be further amended in the course of https://github.com/integritee-network/worker/issues/1242 anyhow. So I left it in the current state.